crawler/plugins/emitters/sas_emitter.py

```diff
@@ -0,0 +1,140 @@
+import logging
+import os
+import json
+import time
+
+import requests
+
+from iemit_plugin import IEmitter
+from plugins.emitters.base_http_emitter import BaseHttpEmitter
+
+logger = logging.getLogger('crawlutils')
+
+
+class SasEmitter(BaseHttpEmitter, IEmitter):
+
+    def get_emitter_protocol(self):
+        return 'sas'
+
+    def emit(self, frame, compress=False,
+             metadata={}, snapshot_num=0, **kwargs):
+        """
+
+        :param frame: a frame containing extracted features
+        :param compress:
+        :param metadata:
+        :param snapshot_num:
+        :return: None
+        """
+        self.token_filepath = kwargs.get("token_filepath", "")
+        self.access_group_filepath = kwargs.get("access_group_filepath", "")
+        self.cloudoe_filepath = kwargs.get("cloudoe_filepath", "")
+
+        iostream = self.format(frame)
+        if compress:
+            proto = self.get_emitter_protocol()
+            raise NotImplementedError(
+                '%s emitter does not support gzip.' % proto
+            )
+        if self.emit_per_line:
+            iostream.seek(0)
+            for line in iostream.readlines():
+                self.post(line, metadata)
+        else:
+            self.post(iostream.getvalue(), metadata)
+
+    '''
+    This function retrievs sas token information from k8s secrets.
+    Current model of secret deployment in k8s is through mounting
+    'secret' inside crawler container.
+    '''
+    def get_sas_tokens(self):
+        assert(os.path.exists(self.token_filepath))
+        assert(os.path.exists(self.access_group_filepath))
+        assert(os.path.exists(self.cloudoe_filepath))
+
+        fp = open(self.access_group_filepath)
+        access_group = fp.read().rstrip('\n')
+        fp.close()
+
+        fp = open(self.cloudoe_filepath)
+        cloudoe = fp.read().rstrip('\n')
+        fp.close()
+
+        fp = open(self.token_filepath)
+        token = fp.read().rstrip('\n')
+        fp.close()
+
+        return(token, cloudoe, access_group)
+
+    '''
+    SAS requires following crawl metadata about entity
+    being crawled.
+        - timestamp
+        - namespace
+        - features
+        - source type
+    This function parses the crawled metadata feature and
+    gets these information.
+    '''
+    def __parse_crawl_metadata(self, content=''):
+        if self.emit_format == 'csv':
```

Comment:
If you don't want to support anything other than csv, better to catch it inside init() itself as is done by crawler/plugins/emitters/mtgraphite_emitter.py. 
This way you also get rid of additional self.emit_format var
Otherwise, init() says sas plugin supports 4 formats, and here it says only csv.